backend/internal/flow/engine/flowengine.go

```diff
@@ -0,0 +1,233 @@
+			currentNode, err = e.resolveToNextNode(ctx.Graph, currentNode)
+			if err != nil {
+				return flowStep, errors.New("error moving to next node: " + err.Error())
+			}
+			ctx.CurrentNode = currentNode
+			continue
+		} else if nodeResp.Status == constants.FlowStatusIncomplete {
+			// If the node returns incomplete status, set the flow step details and return.
+			// The same node will be executed again in the next request with the required data.
+			if nodeResp.Type == constants.FlowStepTypeRedirection {
+				err := e.resolveStepForRedirection(nodeResp, &flowStep)
+				if err != nil {
+					return flowStep, errors.New("error resolving step for redirection: " + err.Error())
+				}
+
+				return flowStep, nil
+			} else if nodeResp.Type == constants.FlowStepTypeView {
+				err := e.resolveStepDetailsForPrompt(nodeResp, &flowStep)
+				if err != nil {
+					return flowStep, errors.New("error resolving step details for prompt: " + err.Error())
+				}
+
+				return flowStep, nil
+			} else {
+				return flowStep, errors.New("unsupported response type returned from the node: " + nodeResp.Type)
+			}
+		} else if nodeResp.Status == constants.FlowStatusPromptOnly {
+			// If it is a prompt only node, set to the next node and return the current flow step.
+			// The next node will be executed in the next request with the requested data.
+			currentNode, err = e.resolveToNextNode(ctx.Graph, currentNode)
+			if err != nil {
+				return flowStep, errors.New("error moving to next node: " + err.Error())
+			}
+			ctx.CurrentNode = currentNode
+			err := e.resolveStepDetailsForPrompt(nodeResp, &flowStep)
+			if err != nil {
+				return flowStep, errors.New("error resolving step details for prompt: " + err.Error())
+			}
+
+			return flowStep, nil
+		} else if nodeResp.Status == constants.FlowStatusError {
+			// If the node returns an error status, set the flow step status to error and return.
+			flowStep.Status = constants.FlowStatusError
+			if nodeResp.Error != "" {
+				return flowStep, errors.New("error returned from the node: " + nodeResp.Error)
+			}
+			return flowStep, errors.New("error returned from the node")
+		} else {
+			// If the node returns an unsupported status, return an error.
+			return flowStep, errors.New("unsupported status returned from the node: " + nodeResp.Status)
+		}
+	}
+
+	// If we reach here, it means the flow has been executed successfully.
+	flowStep.Status = constants.FlowStatusComplete
+
+	// If the current node response has an assertion, set it in the flow step.
+	if ctx.CurrentNodeResponse != nil && ctx.CurrentNodeResponse.Assertion != "" {
+		flowStep.Assertion = ctx.CurrentNodeResponse.Assertion
+	}
+
+	return flowStep, nil
+}
+
+func (e *FlowEngine) resolveToNextNode(graph model.GraphInterface,
+	currentNode model.NodeInterface) (model.NodeInterface, error) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "FlowEngine"))
+
+	nextNodeID := currentNode.GetNextNodeID()
```

Comment:
This can have multiple nodes.. Need to improve this method impl.. Probably need to include more input params too.